node/service.go

```diff
@@ -0,0 +1,11 @@
+package node
+
+import "context"
+
+type Service interface {
+	// Run starts the service and returns an error only if service fails to start
+	// Run does not block and can be canceled with the given context
+	Run(ctx context.Context) error
+	// Wait waits for the service to shut down gracefully
+	Wait()
+}
```

Comment:
This should be moved to root level so that we signify that anyone can implement this

Reply:
done